subworkflows/local/prepare_genome/main.nf

```diff
@@ -342,4 +367,8 @@ workflow PREPARE_GENOME {
     salmon_index     = ch_salmon_index           // channel: path(salmon/index/)
     kallisto_index   = ch_kallisto_index         // channel: [ meta, path(kallisto/index/) ]
     versions         = ch_versions.ifEmpty(null) // channel: [ versions.yml ]
+
+    publish:
+    ch_genome       >> (params.save_reference ? 'genome' : null)
+    ch_genome_index >> (params.save_reference ? 'genome/index' : null)
```

Comment:
Here's one way to shorten the publish section -- gather related channels for each publish target in the main body

Reply:
Was going to suggest this for `align_star`, `quantify_rsem` etc, think that's the right thing to do.

But it does feel a bit like we're doing something publishing-specific in the main body of the workflow.

Could we do like:

```
    publish:
        Channel.empty().mix(
            ch_splicesites,
            ch_bbsplit_index,
            ch_star_index,
            ch_rsem_index,
            ch_hisat2_index,
            ch_salmon_index,
            ch_kallisto_index,
        ) >>  (params.save_reference ? 'genome/index' : null)
```
?

Reply:
yes that also works